backend/src/layer/data/image_validation/api_request.rs

```diff
@@ -0,0 +1,115 @@
+    /// These results are provided in a json, which will be returned if nothing went wrong.
+    /// # Params
+    /// `b64_image`<br>
+    /// This param contains the image as string.
+    /// # Errors
+    /// If the api responded with an error or any connection fault happened, an error will be returned.
+    /// See [`crate::interface::image_validation::ImageValidationError`] for more info about the errors.
+    /// # Return
+    /// The mentioned json ([`SafeSearchJson`]), containing the evaluated values.
+    pub async fn encoded_image_validation(&self, b64_image: String) -> Result<SafeSearchJson> {
+        let token = self.auth_config.generate_auth_token(TOKEN_LIFETIME).await?;
+        let json_resp = self.request_api(b64_image, token).await?.responses.pop();
+        match json_resp {
+            None => Err(InvalidResponse),
+            Some(json) => Ok(json.safeSearchAnnotation),
+        }
+    }
+
+    async fn request_api(
+        &self,
+        b64_image: String,
+        auth_token: String,
+    ) -> Result<SafeSearchResponseJson> {
+        let resp = reqwest::Client::new()
+            .post(API_REST_URL)
+            .header(
+                reqwest::header::AUTHORIZATION,
+                format!("Bearer {auth_token}"),
+            )
+            .header(PROJECT_ID_HEADER, &self.google_project_id)
+            .header(reqwest::header::CONTENT_TYPE, CONTENT_TYPE)
+            .header(reqwest::header::ACCEPT_CHARSET, CHARSET)
+            .body(build_request_body(&b64_image))
+            .send()
+            .await?;
+        // TODO retry with error json if response could not be decoded.
+        // TODO For now, this decode error (containing the response error json)..
+        // TODO ..will be displayed as decode error and not as api error.
+        Ok(resp.json::<SafeSearchResponseJson>().await?)
+    }
+}
+
+fn build_request_body(b64_image: &str) -> String {
+    format!(
+        r#"{{"requests":[{{"image":{{"content":"{b64_image}"}},"features":[{{"type":"{REQUEST_TYPE}"}},]}}]}}"#
+    )
+}
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::unwrap_used)]
+    use crate::layer::data::image_validation::api_request::ApiRequest;
+
+    // Very Small b64 image
+    static B64_IMAGE: &str = "iVBORw0KGgoAAAANSUhEUgAAAAgAAAAIAQMAAAD+wSzIAAAABlBMVEX///+/v7+jQ3Y5AAAADklEQVQI12P4AIX8EAgALgAD/aNpbtEAAAAASUVORK5CYII";
+    static JSON_PATH: &str = "src/layer/data/image_validation/test/test-client.json";
+
+    #[tokio::test]
+    #[ignore]
+    async fn test_generate_auth_token() {
+        let api_req = ApiRequest::new(String::from(JSON_PATH), String::from("mensaka")).unwrap();
+        let resp = api_req
+            .encoded_image_validation(String::from(B64_IMAGE))
+            .await;
+        assert!(resp.is_ok());
+        println!("{resp:?}");
+    }
+
+    // Tests are only possible if we get the client.json hidden in the ci
```

Comment:
should work now, the according env var is set with a path. but i have not tried it.
this can be tested easyly when reading the file in the config reader class.